src/aws-redshiftserverless/namespace.ts

```diff
@@ -0,0 +1,378 @@
+   *
+   * @default - no IAM role associated
+   */
+  readonly iamRoles?: IRole[];
+
+  /**
+   * A Customer Managed Key used to encrypt your data.
+   *
+   * @default - use AWS managed key
+   */
+  readonly kmsKey?: IKey;
+
+  /**
+   * The types of logs the namespace can export.
+   *
+   * @default - no logs export
+   */
+  readonly logExports?: LogExport[];
+
+  /**
+   * The namespace name.
+   *
+   * @default - auto generate
+   */
+  readonly namespaceName?: string;
+}
+
+/**
+ * The types of logs the namespace can export.
+ */
+export enum LogExport {
+  /**
+   * User log
+   */
+  USER_LOG = 'userlog',
+
+  /**
+   * Connection log
+   */
+  CONNECTION_LOG = 'connectionlog',
+
+  /**
+   * User activity log
+   */
+  USER_ACTIVITY_LOG = 'useractivitylog',
+}
+
+/**
+ * Attributes for importing a Redshift Serverless Namespace.
+ */
+export interface NamespaceAttributes {
+  /**
+   * The namespace name
+   */
+  readonly namespaceName: string;
+  /**
+   * The namespace id
+   */
+  readonly namespaceId: string;
+}
+
+/**
+ * Represents a Redshift Serverless Namespace construct in AWS CDK.
+ *
+ * @example
+ *
+ * const nameSpace = new Namespace(
+ *   stack,
+ *   'Namespace',
+ *   {
+ *     namespaceName: 'my-namespace',
+ *   },
+ * );
+ */
+export class Namespace extends Resource implements INamespace {
+  /**
+   * Imports an existing Namespace from attributes
+   */
+  public static fromNamespaceAttributes(scope: Construct, id: string, attrs: NamespaceAttributes): INamespace {
+    class Import extends Resource implements INamespace {
+      public readonly namespaceName = attrs.namespaceName;
+      public readonly namespaceId = attrs.namespaceId;
+      public readonly namespaceArn = Stack.of(this).formatArn({
+        resource: 'redshift-serverless',
+        service: 'namespace',
+        resourceName: attrs.namespaceId,
+      });
+    }
+
+    return new Import(scope, id);
+  }
+
+  /**
+   * The namespace Arn
+   */
+  readonly namespaceArn: string;
+  /**
+   * The namespace name
+   */
+  readonly namespaceName: string;
+  /**
+   * The namespace id
+   */
+  readonly namespaceId: string;
+
+  private readonly props: NamespaceProps;
+
+  private readonly iamRoles: IRole[];
+
+  constructor(scope: Construct, id: string, props: NamespaceProps) {
+    super(scope, id, {
+      physicalName:
+        props.namespaceName ??
+        Lazy.string({
+          produce: () => Names.uniqueResourceName(this, { maxLength: 64, allowedSpecialCharacters: '-' }).toLowerCase(),
+        }),
+    });
+    this.props = props;
+    this.iamRoles = props.iamRoles ?? [];
+
+    this.validateAdmin();
+    this.validateDbName();
+    this.validateFinalSnapshot();
+    this.validateDefaultIamRole();
+    this.validateNamespaceName();
+
+    const namespace = this.createResource(this, 'Resource', {
+      adminUsername: this.props.adminUsername,
+      adminUserPassword: this.props.adminUserPassword?.unsafeUnwrap(),
```

Comment:
Should we validate the unwrapped password? (length, etc)

Reply:
I misunderstood. There is a case for using `SecretValue.secretsManager` instead of `SecretValue.unsafePlainText`, so it is fine as it is.